File: 3_aplikacja_GUI/from_python/DeepSeek/InteractiveApp.java
import javafx.application.Application;
import javafx.geometry.Insets;
import javafx.geometry.Pos;
import javafx.scene.Scene;
import javafx.scene.control.*;
import javafx.scene.layout.VBox;
import javafx.stage.Stage;
import javafx.scene.text.Font;
import javafx.scene.paint.Color;
import javafx.scene.text.Text;
import javafx.util.StringConverter;
import javafx.util.converter.IntegerStringConverter;

import java.io.*;
import java.nio.file.Files;
import java.nio.file.Paths;
import java.util.HashMap;
import java.util.Map;

import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;

public class InteractiveApp extends Application {
    private static final String SAVE_FILE = "user_settings.json";
    
    private TextField nameInput;
    private ComboBox<String> colorDropdown;
    private CheckBox emojiCheck;
    private Slider fontSlider;
    private Text resultLabel;
    
    public static void main(String[] args) {
        launch(args);
    }
    
    @Override
    public void start(Stage primaryStage) {
        primaryStage.setTitle("Interactive App");
        
        // Create UI components
        Label nameLabel = new Label("Enter your name:");
        nameInput = new TextField();
        
        Label colorLabel = new Label("Choose a color:");
        colorDropdown = new ComboBox<>();
        colorDropdown.getItems().addAll("Black", "Blue", "Green", "Red", "Purple");
        
        emojiCheck = new CheckBox("Add an emoji \uD83D\uDE0A");
        
        Label fontLabel = new Label("Select font size:");
        fontSlider = new Slider(10, 30, 12);
        fontSlider.setShowTickLabels(true);
        fontSlider.setShowTickMarks(true);
        fontSlider.setMajorTickUnit(5);
        fontSlider.setMinorTickCount(1);
        fontSlider.setSnapToTicks(true);
        
        Button button = new Button("Show Message");
        button.setOnAction(e -> showMessage());
        
        resultLabel = new Text();
        resultLabel.setTextAlignment(javafx.scene.text.TextAlignment.CENTER);
        
        // Layout
        VBox layout = new VBox(10);
        layout.setPadding(new Insets(20));
        layout.setAlignment(Pos.CENTER);
        layout.getChildren().addAll(
            nameLabel, nameInput,
            colorLabel, colorDropdown,
            emojiCheck,
            fontLabel, fontSlider,
            button, resultLabel
        );
        
        // Load saved settings
        loadSettings();
        
        primaryStage.setScene(new Scene(layout, 400, 400));
        primaryStage.show();
    }
    
    private void showMessage() {
        String name = nameInput.getText().trim();
        String color = colorDropdown.getValue();
        boolean emoji = emojiCheck.isSelected();
        int fontSize = (int)fontSlider.getValue();
        
        if (name.isEmpty()) {
            Alert alert = new Alert(Alert.AlertType.WARNING);
            alert.setTitle("Warning");
            alert.setHeaderText(null);
            alert.setContentText("Please enter your name!");
            alert.showAndWait();
            return;
        }
        
        String message = "Hello, " + name + "! Your favorite color is " + color + ".";
        if (emoji) {
            message += " \uD83D\uDE0A";
        }
        
        resultLabel.setText(message);
        resultLabel.setFill(Color.web(color.toLowerCase()));
        resultLabel.setFont(Font.font(fontSize));
        
        saveSettings(name, color, emoji, fontSize);
    }
    
    private void saveSettings(String name, String color, boolean emoji, int fontSize) {
        Map<String, Object> data = new HashMap<>();
        data.put("name", name);
        data.put("color", color);
        data.put("emoji", emoji);
        data.put("font_size", fontSize);
        
        try (Writer writer = new FileWriter(SAVE_FILE)) {
            new Gson().toJson(data, writer);
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
    
    private void loadSettings() {
        if (Files.exists(Paths.get(SAVE_FILE))) {
            try (Reader reader = new FileReader(SAVE_FILE)) {
                Map<String, Object> data = new Gson().fromJson(
                    reader, 
                    new TypeToken<Map<String, Object>>(){}.getType()
                );
                
                nameInput.setText(data.getOrDefault("name", "").toString());
                colorDropdown.setValue(data.getOrDefault("color", "Black").toString());
                emojiCheck.setSelected(Boolean.parseBoolean(data.getOrDefault("emoji", "false").toString()));
                fontSlider.setValue(Double.parseDouble(data.getOrDefault("font_size", "12").toString()));
            } catch (IOException e) {
                e.printStackTrace();
            }
        }
    }
}

File: 3_aplikacja_GUI/from_java/ChatGPT/cpp_translation_chatgpt_fixed/InteractiveApp.cpp
#include "InteractiveApp.h"
#include <QVBoxLayout>
#include <QLabel>
#include <QPushButton>
#include <QMessageBox>
#include <QFont>
#include <QPalette>
#include <fstream>

InteractiveApp::InteractiveApp(QWidget *parent) : QWidget(parent) {
    setWindowTitle("Interactive App");
    setFixedSize(400, 300);

    QVBoxLayout *layout = new QVBoxLayout(this);

    QLabel *nameLabel = new QLabel("Enter your name:");
    nameInput = new QLineEdit();
    layout->addWidget(nameLabel);
    layout->addWidget(nameInput);

    QLabel *colorLabel = new QLabel("Choose a color:");
    colorDropdown = new QComboBox();
    colorDropdown->addItems({"Black", "Blue", "Green", "Red", "Purple"});
    colorDropdown->setCurrentText("Black");
    layout->addWidget(colorLabel);
    layout->addWidget(colorDropdown);

    emojiCheck = new QCheckBox("Add an emoji 😊");
    layout->addWidget(emojiCheck);

    QLabel *fontLabel = new QLabel("Select font size:");
    fontSlider = new QSlider(Qt::Horizontal);
    fontSlider->setRange(10, 30);
    fontSlider->setValue(12);
    layout->addWidget(fontLabel);
    layout->addWidget(fontSlider);

    QPushButton *button = new QPushButton("Show Message");
    layout->addWidget(button);

    resultLabel = new QLabel("");
    resultLabel->setAlignment(Qt::AlignCenter);
    layout->addWidget(resultLabel);

    connect(button, &QPushButton::clicked, this, &InteractiveApp::showMessage);

    loadSettings();
}

void InteractiveApp::showMessage() {
    QString name = nameInput->text().trimmed();
    QString color = colorDropdown->currentText();
    bool emoji = emojiCheck->isChecked();
    int fontSize = fontSlider->value();

    if (name.isEmpty()) {
        QMessageBox::warning(this, "Warning", "Please enter your name!");
        return;
    }

    QString message = "Hello, " + name + "! Your favorite color is " + color + ".";
    if (emoji) {
        message += " 😊";
    }

    resultLabel->setText(message);

    QPalette palette = resultLabel->palette();
    palette.setColor(QPalette::WindowText, QColor(color.toLower()));
    resultLabel->setPalette(palette);

    QFont font = resultLabel->font();
    font.setPointSize(fontSize);
    resultLabel->setFont(font);

    saveSettings(name, color, emoji, fontSize);
}

void InteractiveApp::saveSettings(const QString &name, const QString &color, bool emoji, int fontSize) {
    json data;
    data["name"] = name.toStdString();
    data["color"] = color.toStdString();
    data["emoji"] = emoji;
    data["font_size"] = fontSize;

    std::ofstream file("user_settings.json");
    if (file.is_open()) {
        file << data.dump(4);
    }
}

void InteractiveApp::loadSettings() {
    std::ifstream file("user_settings.json");
    if (!file.is_open()) return;

    json data;
    try {
        file >> data;
    } catch (...) {
        return;
    }

    nameInput->setText(QString::fromStdString(data.value("name", "")));
    colorDropdown->setCurrentText(QString::fromStdString(data.value("color", "Black")));
    emojiCheck->setChecked(data.value("emoji", false));
    fontSlider->setValue(data.value("font_size", 12));
}
